script/Deployer.s.sol

```diff
@@ -0,0 +1,275 @@
+            deployer,
+            type(VanillaProposalValidationStrategy).creationCode,
+            saltNonce
+        );
+        deployments.serialize("VanillaProposalValidationStrategy", vanillaProposalValidationStrategy);
+
+        (address propositionPowerProposalValidationStrategy, ) = noRedeploy(
+            deployer,
+            type(PropositionPowerProposalValidationStrategy).creationCode,
+            saltNonce
+        );
+        deployments.serialize("PropositionPowerProposalValidationStrategy", propositionPowerProposalValidationStrategy);
+
+        (address activeProposalsLimiterProposalValidationStrategy, ) = noRedeploy(
+            deployer,
+            type(ActiveProposalsLimiterProposalValidationStrategy).creationCode,
+            saltNonce
+        );
+
+        deployments.serialize(
+            "ActiveProposalsLimiterProposalValidationStrategy",
+            activeProposalsLimiterProposalValidationStrategy
+        );
+
+        (address propositionPowerAndActiveProposalsLimiterValidationStrategy, ) = noRedeploy(
+            deployer,
+            type(PropositionPowerAndActiveProposalsLimiterValidationStrategy).creationCode,
+            saltNonce
+        );
+
+        deployments.serialize(
+            "PropositionPowerAndActiveProposalsLimiterValidationStrategy",
+            propositionPowerAndActiveProposalsLimiterValidationStrategy
+        );
+
+        // ------- PROXY FACTORY -------
+
+        (address proxyFactory, ) = noRedeploy(deployer, type(ProxyFactory).creationCode, saltNonce);
+        deployments.serialize("ProxyFactory", proxyFactory);
+
+        // ------- SPACE -------
+
+        (address space, ) = noRedeploy(deployer, type(Space).creationCode, saltNonce);
+
+        // If the master space is not initialized, initialize it
+        if (Space(space).owner() == address(0x0)) {
+            // Initializer for the master space, to render it unusable
+            Strategy[] memory emptyStrategyArray = new Strategy[](1);
+            emptyStrategyArray[0] = Strategy(address(0x1), new bytes(0));
+            string[] memory emptyStringArray = new string[](1);
+            emptyStringArray[0] = "";
+            address[] memory emptyAddressArray = new address[](1);
+            emptyAddressArray[0] = address(0x1);
+            Space(space).initialize(
+                InitializeCalldata(
+                    address(0x1),
+                    1,
+                    1,
+                    1,
+                    Strategy(address(0x1), new bytes(0)),
+                    "",
+                    "",
+                    "",
+                    emptyStrategyArray,
+                    emptyStringArray,
+                    emptyAddressArray
+                )
+            );
+        }
+
+        if (Space(space).owner() != address(0x1)) {
+            // Initialization of the master space was frontrun
+            revert SpaceInitializationFailed();
```

Comment:
is the `owner` variable the only variable worth checking ?

Reply:
yeah added a check for proposal validation strategy which prevents proposals from being created. That should prevent all actions in the space then i think